safety/safety.py

```diff
@@ -87,17 +88,31 @@ def fetch_database_url(mirror, db_name, key, cached, proxy):
         if cached_data:
             return cached_data
     url = mirror + db_name
-    r = requests.get(url=url, timeout=REQUEST_TIMEOUT, headers=headers, proxies=proxy)
+
+    try:
+        r = requests.get(url=url, timeout=REQUEST_TIMEOUT, headers=headers, proxies=proxy)
+    except requests.exceptions.ConnectionError:
+        raise NetworkConnectionError()
+    except requests.exceptions.Timeout:
+        raise RequestTimeoutError()
+    except requests.exceptions.RequestException:
+        raise DatabaseFetchError()
+
     if r.status_code == 200:
         data = r.json()
         if cached:
             write_to_cache(db_name, data)
         return data
```

Comment:
As you modified the elif-statements below because of short-circuits, can we bring this to the end of the function without the if-statement? So, the "last case" would be the default, returning the data.